README.lhs

```diff
@@ -0,0 +1,36 @@
+# hspec-junit-formatter
+
+A `JUnit` XML runner/formatter for [`hspec`](http://hspec.github.io/).
+
+<!--
+```haskell
+module Main (main) where
+import Prelude
+import Text.Markdown.Unlit ()
+```
+-->
+
+## Usage
+
+```haskell
+import Test.Hspec
+import Test.Hspec.Core.Runner (defaultConfig, hspecWith)
+import qualified Test.HSpec.JUnit as JUnit
+
+main :: IO ()
+main = do
+  let
+    report = "/tmp/test-results.xml"
+    config = JUnit.configWith report "my-tests" defaultConfig
+
+  hspecWith config spec
+
+spec :: Spec
+spec = describe "Addition" $ do
+  it "adds" $ do
+    2 + 2 `shouldBe` (4 :: Int)
+```
```

Comment:
I'm curious why you moved it away from configuration based use to a `with` pattern?

```hs
main :: IO ()
main = runSpec specs config
config :: Config
config = defaultConfig
  { configFormat = Just $ junitFormat "test-results.xml" "my-tests"
  }
```

Reply:
Hmm, actually I don't see anything wrong with `configWith`. I was thinking of `runJUnitSpec`, which is legacy.

Reply:
It's a good question though still. I looked at `configWith` and `junitFormat` and saw that the latter was built on top of the former, so I figured higher-level would be better for a README introduction. Is that fair?

Reply:
Oh and the `runSpec specs config` part was just wrong. That's just the middle step that gives `IO Summary`, you still need to evaluate it. It's also necessary to respect `getArgs` and ensure (e.g.) `--test-arguments` works.

![](https://files.pbrisbin.com/screenshots/screenshot.339740.png)

(`hspecWith` is exactly that but with a provided, non-default `Config`)

So, you could still keep `junitFormat` but it would have to look like:

```hs
main :: IO ()
main = hspecWith config specs

config :: Config
config = defaultConfig
  { configFormat = Just $ junitFormat "test-results.xml" "my-tests"
  }
```